yarn/src/main/scala/org/apache/spark/deploy/yarn/YarnAllocator.scala

```diff
@@ -86,7 +86,7 @@ private[yarn] class YarnAllocator(
   private var executorIdCounter = 0
   @volatile private var numExecutorsFailed = 0
 
-  @volatile private var targetNumExecutors = args.numExecutors
+  @volatile private var targetNumExecutors = sparkConf.get("spark.executor.instances").toInt
```

Comment:
`sparkConf.getInt()`

Though it's a little weird, because `SparkConf.getInt()` takes a default value. But `spark.executor.instances` should always be set after these changes, so any dummy default should do.